ethereum_history_api/circuits/lib/src/misc/fragment_test.nr

```diff
@@ -185,6 +185,25 @@ mod pop_front {
     }
 }
 
+mod pop_front_array {
+
+    use crate::misc::fragment::Fragment;
+
+    #[test]
+    fn success() {
+        let mut fragment = Fragment::new(1, 2, [1, 2, 3]);
+        assert(fragment.pop_front_array() == [2, 3]);
+        assert(fragment.length == 0);
+    }
+
+    #[test(should_fail_with="Cannot pop from an empty fragment")]
+    fn fail() {
+        let mut fragment = Fragment::new(1, 1, [1, 2, 3]);
+        assert(fragment.pop_front_array() == [2]);
+        assert(fragment.pop_front_array() == [3]);
+    }
```

Comment:
I would add a test with this fragment but for example:
let _: [u8; 2] = fragment.pop_front_array();
If I understand correctly it should fail, but the message that it would return may be misleading.